### PR TITLE
Sanitize sensitive fields in http.query()

### PR DESCRIPTION
**Pull request by techhat**

### CODE REVIEW

**Comment by jfindlay** [comment]
Nice, thanks @techhat.

**Comment by jfindlay** [comment]
Go Go Jenkins!

**Comment by thatch45** [comment]
Is this code in 2015.8?

**Comment by techhat** [comment]
@thatch45, no, but there's also no code in 2015.8 that would make use of it.

**Comment by cachedout** [comment]
This broke a bunch of tests. Fixing.

**Comment by techhat** [comment]
@cachedout let me know what I can do.

File: salt/utils/http.py [hunk 1]
```diff
@@ -120,6 +120,7 @@ def query(url,
           stream=False,
           handle=False,
           agent=USERAGENT,
+          hide_fields=None,
```
Comment (bechtoldt) [inline salt/utils/http.py]: This is breaking (almost) all salt.utils.http.query() calls where ``hide_fields`` isn't overriden. fixed by #28294

**Pull request merged**